tests/unit_tests/scheduler/test_job.py

```diff
@@ -0,0 +1,228 @@
+import asyncio
+import json
+import shutil
+from typing import List
+from unittest.mock import AsyncMock, MagicMock
+
+import pytest
+
+import ert
+from ert.ensemble_evaluator._builder._realization import Realization
+from ert.load_status import LoadResult, LoadStatus
+from ert.run_arg import RunArg
+from ert.scheduler import Scheduler
+from ert.scheduler.driver import JobEvent
+from ert.scheduler.job import STATE_TO_LEGACY, Job, State
+
+
+def create_scheduler():
+    sch = MagicMock()
+    sch._events = asyncio.Queue()
+    sch.driver = AsyncMock()
+    return sch
+
+
+@pytest.fixture
+def realization():
+    run_arg = RunArg(
+        run_id="",
+        ensemble_storage=MagicMock(),
+        iens=0,
+        itr=0,
+        runpath="test_runpath",
+        job_name="test_job",
+    )
+    realization = Realization(
+        iens=run_arg.iens,
+        forward_models=[],
+        active=True,
+        max_runtime=None,
+        run_arg=run_arg,
+        num_cpu=1,
+        job_script=str(shutil.which("job_dispatch.py")),
+    )
+    return realization
+
+
+async def assert_scheduler_events(
+    scheduler: Scheduler, job_events: List[JobEvent]
+) -> None:
+    await asyncio.sleep(0)
+    assert scheduler._events.qsize() == len(job_events)
+    for job_event in job_events:
+        output = json.loads((await scheduler._events.get()).decode("utf-8"))
+        event = output.get("data").get("queue_event_type")
+        assert event == STATE_TO_LEGACY[job_event.value]
+    # should be no more events
+    assert scheduler._events.empty()
+    await asyncio.sleep(0)
+
+
+def clear_queue(queue: asyncio.Queue) -> None:
+    print(f"Clearing {queue.qsize()} items")
+    while not queue.empty():
+        queue.get_nowait()
+
+
+@pytest.mark.asyncio
+async def test_job_acquires_semaphore(realization):
+    scheduler = create_scheduler()
+    job = Job(scheduler, realization)
+    sem = asyncio.Semaphore(1)
+    start_event = asyncio.Event()
+    asyncio.create_task(job(start_event, sem))
+    await asyncio.sleep(0)
+    start_event.set()
+    await asyncio.sleep(0)
+    assert sem.locked()
+
+
+@pytest.mark.asyncio
+async def test_job_waits_for_start_event(realization):
```

Comment:
Hmm, are we testing implementation details again? In Scheduler we do:
```
            for iens, job in self._jobs.items():
                self._tasks[iens] = asyncio.create_task(
                    job(start, sem, self._max_submit)
                )

            start.set()
```
and I am not really sure exactly why we do it, but maybe we should test instead on what bad things happen why do don't have this extra wait.